cylc/flow/task_pool.py

```diff
@@ -1002,53 +1008,106 @@ def report_unmet_deps(self):
-        itasks, bad_items = self.filter_task_proxies(items)
-        to_queue = []
+        # Release active tasks:
+        itasks, unmatched = self.filter_task_proxies(items, warn=False)
         for itask in itasks:
-            if itask.state.reset(is_held=False):
-                self.data_store_mgr.delta_task_held(itask)
-                if (
-                        all(itask.is_ready_to_run())
-                        and not itask.state.is_runahead
-                ):
-                    to_queue.append(itask)
-        self.queue_tasks(to_queue)
-        return len(bad_items)
+            self.release_held_active_task(itask)
+        # Unhold future tasks:
+        n_warnings, task_items = self._explicit_match_tasks_to_hold(unmatched)
+        self.tasks_to_hold.difference_update(task_items)
+        self.workflow_db_mgr.put_tasks_to_hold(self.tasks_to_hold)
+        LOG.debug(f"Tasks to hold: {self.tasks_to_hold}")
+        return n_warnings
 
     def release_hold_point(self) -> None:
-        """Release all tasks and unset the workflow hold point."""
+        """Unset the workflow hold point and release all held active tasks."""
         self.hold_point = None
-        to_queue = []
         for itask in self.get_all_tasks():
-            if itask.state.reset(is_held=False):
-                self.data_store_mgr.delta_task_held(itask)
-                if (
-                        all(itask.is_ready_to_run())
-                        and not itask.state.is_runahead
-                ):
-                    to_queue.append(itask)
-        self.queue_tasks(to_queue)
+            self.release_held_active_task(itask)
+        self.tasks_to_hold.clear()
+        self.workflow_db_mgr.put_tasks_to_hold(self.tasks_to_hold)
         self.workflow_db_mgr.delete_workflow_hold_cycle_point()
 
+    def _explicit_match_tasks_to_hold(
+        self, items: Iterable[str]
+    ) -> Tuple[int, Set[Tuple[str, 'PointBase']]]:
+        """Match future tasks based on the assumption that no active tasks
+        matched the items.
+
+        Returns (n_warnings, matched_tasks)
+        """
+        n_warnings = 0
+        matched_tasks: Set[Tuple[str, 'PointBase']] = set()
+        for item in items:
+            point_str, name_str, status = self._parse_task_item(item)
+            if status or ('*' in item) or ('?' in item) or ('[' in item):
+                # Glob or task state was not matched in active pool
+                LOG.warning(f"No active tasks matching: {item}")
+                n_warnings += 1
+                continue
+            taskdefs = self.config.find_taskdefs(name_str)
+            if not taskdefs:
+                LOG.warning(self.ERR_TMPL_NO_TASKID_MATCH.format(name_str))
+                n_warnings += 1
+                continue
+            if point_str is None:
+                LOG.warning(f"No active instances of task: {item}")
+                n_warnings += 1
+                continue
```

Comment:
This function is matching future tasks not active ones?

Reply:
It's a bit confusing I know. This function relies on you having previously filtered out the matching active tasks using `self.filter_task_proxies()`. So if the point is not specified e.g. `mytask`, equivalent to `mytask.*`, and there is no active instance of `mytask`, then I say that is the correct warning to give.